qubesmanager/qube_manager.py

```diff
@@ -61,140 +71,397 @@ def mousePressEvent(self, e):  # pylint: disable=invalid-name
         :return: None
         """
         try:
-            self.info_widget.update_vm_state()
+            self.state['power'] = self.vm.get_power_state()
+
+            if self.vm.is_running():
+                if hasattr(self.vm, 'template') and \
+                        self.vm.template.is_running():
+                    self.state['outdated'] = "to-be-outdated"
+                else:
+                    for vol in self.vm.volumes.values():
+                        if vol.is_outdated():
+                            self.state['outdated'] = "outdated"
+                            break
+            else:
+                self.state['outdated'] = ""
+
+            if self.vm.klass in {'TemplateVM', 'StandaloneVM'} and \
+                    self.vm.features.get('updates-available', False):
+                self.state['outdated'] = 'update'
+
             if not event or event.endswith(':label'):
-                self.label_widget.update()
+                self.label = self.vm.label
             if not event or event.endswith(':template'):
-                self.template_widget.update()
+                try:
+                    self.template = self.vm.template.name
+                except AttributeError:
+                    self.template = None
             if not event or event.endswith(':netvm'):
-                self.netvm_widget.update()
+                self.netvm = getattr(self.vm, 'netvm', None)
+                if self.netvm:
+                    self.netvm = self.netvm.name
+                else:
+                    self.netvm = "n/a"
+                if self.qid != 0 and self.vm.property_is_default("netvm"):
+                    self.netvm = "default (" + self.netvm + ")"
             if not event or event.endswith(':internal'):
                 # this is a feature, not a property; TODO: fix event handling
-                self.internal_widget.update()
+                self.internal = self.vm.features.get('internal', False)
             if not event or event.endswith(':ip'):
-                self.ip_widget.update()
+                self.ip = getattr(self.vm, 'ip', "n/a")
             if not event or event.endswith(':include_in_backups'):
-                self.include_in_backups_widget.update()
+                self.inc_backup = getattr(self.vm, 'include_in_backups', None)
             if not event or event.endswith(':backup_timestamp'):
-                self.last_backup_widget.update()
+                self.last_backup = getattr(self.vm, 'backup_timestamp', None)
+                if self.last_backup:
+                    self.last_backup = str(datetime.fromtimestamp(
+                        self.last_backup))
             if not event or event.endswith(':default_dispvm'):
-                self.dvm_template_widget.update()
+                self.dvm = getattr(self.vm, 'default_dispvm', None)
+                if self.vm.property_is_default("default_dispvm"):
+                    self.dvm = "default (" + self.dvm.name + ")"
```

Comment:
Hmm, `self.vm` is a `VmInfo` object, not `QubesVM`, right? Then, this would require adding `__str__` method.

BTW, in the meantime `QubesVM` object in `qubesadmin` module got caching support directly (including reacting to events to update it), so you can use that directly now.